### PR TITLE
Specify mirrord binary in settings

### LINKED ISSUE #6: allow specifying mirrord binary in extension settings
Right now it always tries to find the binary by itself, and if it can't download or use any of the local found binaries it fails.
Let users specify a specific binary to use, so then they can use it without downloading or let them pin to a specific binary.
Would be useful for extension development and testing as well

### PR DESCRIPTION
**Pull request by Razz4780**
Closes #6

### CODE REVIEW

File: src/binaryManager.ts [hunk 1]
```diff
@@ -53,10 +54,53 @@ export async function getLocalMirrordBinary(version?: string): Promise<string |
     return null;
 }
 
+async function getConfiguredMirrordBinary(): Promise<string | undefined> {
+    const configured = workspace.getConfiguration().get<string | null>("mirrord.binaryPath");
+    if (!configured) {
+        return undefined;
```
Comment (eyalb181) [inline src/binaryManager.ts:60]: I think `null` would be more correct here (since it's not undefined behavior)

File: src/binaryManager.ts [hunk 2]
```diff
@@ -53,10 +54,53 @@ export async function getLocalMirrordBinary(version?: string): Promise<string |
...
+    try {
+        version = await new MirrordAPI(configured).getBinaryVersion();
+        if (version === undefined) {
+            throw new Error("version command returned malformed output");
+        }
+    } catch (err) {
+        new NotificationBuilder()
+            .withMessage(`failed to used mirrord binary specified in settings due to failed version check: ${err}`)
+            .warning();
+        return undefined;
```
Comment (eyalb181) [inline src/binaryManager.ts:73]: Same as above

File: src/binaryManager.ts [hunk 3]
```diff
@@ -53,10 +54,53 @@ export async function getLocalMirrordBinary(version?: string): Promise<string |
...
+async function getConfiguredMirrordBinary(): Promise<string | undefined> {
+    const configured = workspace.getConfiguration().get<string | null>("mirrord.binaryPath");
+    if (!configured) {
+        return undefined;
+    }
+
+    let version;
+    try {
+        version = await new MirrordAPI(configured).getBinaryVersion();
+        if (version === undefined) {
```
Comment (eyalb181) [inline src/binaryManager.ts:66]: Can't you do `if !version`?

**Review by eyalb181 — CHANGES_REQUESTED** [review]

File: src/binaryManager.ts (hunk 3 above)
Comment (Razz4780) [reply src/binaryManager.ts]: Sure!

**Pull request merged**